src/codegen/dispatchers/map_dispatcher.cpp

```diff
@@ -0,0 +1,45 @@
+#include "sdfg/codegen/dispatchers/map_dispatcher.h"
+
+#include "sdfg/codegen/dispatchers/sequence_dispatcher.h"
+
+namespace sdfg {
+namespace codegen {
+
+MapDispatcher::MapDispatcher(LanguageExtension& language_extension, Schedule& schedule,
+                             structured_control_flow::Map& node, Instrumentation& instrumentation)
+    : NodeDispatcher(language_extension, schedule, node, instrumentation), node_(node) {
+
+      };
+
+void MapDispatcher::dispatch_node(PrettyPrinter& main_stream, PrettyPrinter& globals_stream,
+                                  PrettyPrinter& library_stream) {
+    LoopSchedule loop_schedule = this->schedule_.loop_schedule(&this->node_);
+    if (loop_schedule == LoopSchedule::VECTORIZATION) {
+        main_stream << "#pragma omp simd" << std::endl;
+    }
+    if (loop_schedule == LoopSchedule::MULTICORE) {
```

Comment:
This is the right way to go, but we have a OMP and Highway Scheduler for For-Loops right now. Let's ignore the loop_schedule set in this PR and we need a second PR to completely re-think the schedule property and dispatchers for the different schedules